r/R/utilities_expression.R

```diff
@@ -203,7 +209,7 @@ getStats <- function(data) {
 sparsify <- function(expression) {
   data.table::setnafill(expression, fill = 0)
   sparse_matrix <-
-    Matrix::Matrix(Matrix::as.matrix(expression), sparse = T)
+    as(as.matrix(expression), 'dgCMatrix')
```

Comment:
yesss that conversion was horrible.

Reply:
Changed this because some of the tests were creating a `dTiMatrix` or something like that which led to errors. This just makes it explicit the type of sparse matrix we want.

Reply:
yep, newer matrix converts (mathematically) diagonal matrices to that diagonal matrix type. The other solution is to add `diagonal = F` to the Matrix::Matrix call, but this is much nicer.

Reply:
as okay cool -- thanks for the heads up. We do have a few other calls of `Matrix::Matrix(..., sparse = T)`. They don't seem to be causing any errors in our tests but maybe they should be updated as well to prevent `dTiMatrix` shenanigans